tools/eubfr-cli/bin/cli.js

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env node
+
+const program = require('commander');
+const pkg = require('../package.json');
+
+// Utilities
+const getCredentials = require('../lib/getProducerCredentials');
+const getAllProducers = require('../lib/getAllProducers');
+
+// Commands
+const uploadFiles = require('../commands/upload');
+const showFile = require('../commands/show');
+const deleteFiles = require('../commands/delete');
+
+program.version(pkg.version).usage('[command] [option]');
+
+program
+  .command('upload [files...]')
+  .description('Uploads content to the data lake.')
+  .option('-p, --producer [producer]', "Producer's name.")
+  .action((files, options) => {
+    let credentials = [];
+    // If -p is passed without actual value, it will be boolean true
+    // Which is useless information in our case
+    const producerIsSet =
+      options.producer &&
+      options.producer !== '' &&
+      typeof options.producer !== 'boolean';
+    const filesAreSet = files.length !== 0;
+
+    if (!producerIsSet && filesAreSet) {
+      console.log('Please provide a producer with -p flag and value after it.');
+      process.exit(1);
+    }
+
+    if (producerIsSet) {
+      const { producer } = options;
+      credentials = [{ [producer]: getCredentials(producer) }];
+    }
+
+    if (!producerIsSet && !filesAreSet) {
+      console.log('All files for all producers will be uploaded.');
+      const producers = getAllProducers();
+      credentials = producers.map(producer => ({
+        [producer]: getCredentials(producer),
+      }));
+    }
+
+    uploadFiles({ files, credentials });
+  });
+
+program
+  .command('show [file]')
+  .description('Displays files of a given producer.')
+  .option('-p, --producer [producer]', "Producer's name. Defaults to 'agri'.")
+  .action((file, options) => {
+    const producer = options.producer || 'agri';
```

Comment:
Why setting the default producer to agri?
This may cause unwanted political questions.

We may check if the producer is empty and trigger a warning message.

Reply:
It's our first ETL and it's also first alphabetically :)
I agree it's not too good of an argument to keep it though, will add a warning message.